### ISSUE TITLE
Athena dialect does not support columns prefixed with underscore

### ISSUE THREAD

**Issue by jessedobbelaere** [sqlfluff/sqlfluff#4665]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A sql file fails to parse using SQLfluff when using the Athena dialect with an underscore-prefixed column identifier, like `_row` or `_fivetran_synced`

### Expected Behaviour

Sqlfluff can parse a model when referencing underscore-prefixed column names

### Observed Behaviour

A dbt model like this fails using the athena dialect:

```sql
SELECT 
    _fivetran_synced,
    quarter
FROM ...
```

```
== [/Users/XXX/XXX.sql] FAIL                              
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'SELECT\n 
                       | _fivetran_synced,\n    quarter...'
```

It works with the `ansi` dialect. Leaving out the `_fivetran_synced` column fixes the error. 

Athena supports columns prefixed with underscore: https://docs.aws.amazon.com/athena/latest/ug/tables-databases-columns-names.html#names-that-begin-with-an-underscore but the Sqlfluff dialect doesn't seem to support it?

Seems related to #3678 and #3965 which only added support for underscores in aliases?

### How to reproduce

See above

### Dialect

athena

### Version

```
sqlfluff --version
sqlfluff, version 2.0.2

python --version
Python 3.10.10
```

### Configuration

```
# See https://docs.sqlfluff.com/en/stable/rules.html
[sqlfluff]
dialect = athena
max_line_length = 140

# Disable certain rules here.
# structure.column_order: Select wildcards then simple targets before calculations and aggregates.
# references.keywords: Keywords should not be used as identifiers (e.g. `AS user`)
exclude_rules = structure.column_order,references.keywords

[sqlfluff:templater:jinja]
load_macros_from_path = macros

# We can use "templater = dbt" for better support of dbt macro's, but then SQLFluff triggers a dbt compile job
# which takes a long time and needs assume-it. It also does not show any linter output in vscode (bug?).
# For now, we use the more basic jinja templater, but we have to mock dbt-specific macro's below so the linter does not fail.
[sqlfluff:templater:jinja:macros]
dbt_env_var = {% macro env_var(name, fallback) %}item{% endmacro %}
var = {% macro var(name) %}123{% endmacro %}

# Disable "Line is too long" only for comments
[sqlfluff:rules:layout.long_lines]
ignore_comment_lines = True

# SQL keywords should be uppercased
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

# SQL functions should be uppercased
[sqlfluff:rules:capitalisation.functions]
capitalisation_policy = upper
extended_capitalisation_policy = upper

# Use unquoted keyword identifiers (e.g. AS type)
[sqlfluff:rules:capitalisation.identifiers]
unquoted_identifiers_policy = none

```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Was fixed with #4636
